src/containers/MainScreen/index.js

```diff
@@ -8,7 +8,7 @@ import { logout } from 'actions/userActions';
 import translate from 'utils/i18n';
 import styles from './styles';
 
-const MainScreen = ({ email, logout }) => (
+const MainScreen = ({ user: { email }, logout }) => (
```

Comment:
Why are you passing all the user object if you just need the email?

Reply:
Because previously when the user logout and the component re renders. The selector tries to get the email value of an empty object and this was returning undefined, causing to throw the warning that email was required but its value is undefined. 

Sending the user object prevents the warning because the object will not be undefined just empty.

Reply:
In that case, you should just set the prop as not required. Wrapping the prop with an object makes useless the prop type checking.